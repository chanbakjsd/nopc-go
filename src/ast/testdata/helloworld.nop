//This is what Hello World would look like in Noplang.
//I would really question life if something breaks the compilation of this file.
//Maybe you should if you are the person who broke compilation of this file.

fn main() {
	println("Hello World!")
}
