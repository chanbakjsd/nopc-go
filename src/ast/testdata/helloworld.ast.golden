{
	"Source": "testdata/helloworld.nop",
	"CompilerPragma": [],
	"Import": null,
	"Structs": null,
	"Traits": null,
	"Impls": null,
	"Globals": null,
	"Constants": null,
	"Functions": [
		{
			"Name": "main",
			"Parameters": [],
			"ReturnType": {
				"Name": "",
				"ToInferType": false,
				"InnerType": null,
				"Public": false,
				"Static": false,
				"Mutable": false
			},
			"Body": [
				{
					"Type": "[EvaluatedStatement]",
					"Actual": {
						"Type": "[FunctionCall]",
						"ResolvedType": {
							"Name": "",
							"ToInferType": true,
							"InnerType": null,
							"Public": false,
							"Static": false,
							"Mutable": false
						},
						"LiteralValue": null,
						"Arguments": [
							{
								"Type": "[PropertyName]",
								"ResolvedType": {
									"Name": "",
									"ToInferType": true,
									"InnerType": null,
									"Public": false,
									"Static": false,
									"Mutable": false
								},
								"LiteralValue": "println",
								"Arguments": null
							},
							{
								"Type": "[Literal]",
								"ResolvedType": {
									"Name": "ustring",
									"ToInferType": false,
									"InnerType": null,
									"Public": false,
									"Static": false,
									"Mutable": false
								},
								"LiteralValue": "Hello World!",
								"Arguments": null
							}
						]
					}
				}
			]
		}
	]
}